utils/ccdatofhir/mapping_resource_immunization.bal

```diff
@@ -0,0 +1,262 @@
+            if statusReason is r4:CodeableConcept {
+                immunization.statusReason = statusReason;
+            }
+        } else if typeCode is string && typeCode == "MFST" {
+            xml entryRelationshipObservationElement = entryRelationship/<v3:observation|observation>;
+            xml reference = entryRelationshipObservationElement/<v3:text|text>/<v3:reference|reference>;
+            string|error? referenceValue = reference.value;
+            if referenceValue is string {
+                r4:ImmunizationReaction immunizationReaction = {
+                    detail: {
+                        reference: referenceValue
+                    }
+                };
+                immunization.reaction = [immunizationReaction];
+            }
+        }
+
+        xml codeElement = entryRelationship/<v3:act|act>/<v3:code|code>;
+        string|error? code = codeElement.code;
+        if code is string {
+            if code == "48767-8" {
+                xml textElement = entryRelationship/<v3:act|act>/<v3:text|text>;
+                string|error? text = textElement.data();
+                if text is string {
+                    immunization.note = [{text: text}];
+                }
+            }
+        }
+
+        return immunization;
+    }
+    return ();
+}
+
+isolated function mapCcdaRefusalReasonToFhirStatusReason(xml codingElement) returns r4:CodeableConcept? {
+    if isXMLElementNotNull(codingElement) {
+        r4:CodeableConcept mapCcdaCodingtoFhirCodeableConceptResult = {};
+        string|error? codeVal = codingElement.code;
+        if codeVal is string {
+            string? code;
+            match codeVal {
+                "IMMUNE" => {
+                    code = "IMMUNE";
+                }
+                "MEDPREC" => {
+                    code = "MEDPREC";
+                }
+                "OSTOCK" => {
+                    code = "OSTOCK";
+                }
+                "PATOBJ" => {
+                    code = "PATOBJ";
+                }
+                _ => {
+                    code = ();
+                }
+            }
+            mapCcdaCodingtoFhirCodeableConceptResult = {
+                coding: [
+                    {
+                        code: code
+                    }
+                ]
+            };
+        }
+
+        string|error? codeSystem = codingElement.codeSystem;
+        if codeSystem is string {
+            mapCcdaCodingtoFhirCodeableConceptResult.coding[0].system = codeSystem;
+        }
+
+        string|error? displayName = codingElement.displayName;
+        if displayName is string {
+            mapCcdaCodingtoFhirCodeableConceptResult.coding[0].display = displayName;
+        }
+        return mapCcdaCodingtoFhirCodeableConceptResult;
+    }
+    return ();
+}
+
+isolated function mapCcdaStatusCodeToFhirImmunizationStatus(xml codingElement) returns r4:ImmunizationStatus {
+    string|error? codeVal = codingElement.data();
```

Comment:
```suggestion
    string codeVal = codingElement.data();
```
You don't need `error?`. Please check what the return type of the data() function in this case. Use only the type that are necessary for right hand side

Reply:
Check for similar other places

Reply:
Updated in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/e973cad61936b398a3651cbc34f89d9f66c2aa15